protocol-spec/generic.md

```diff
@@ -134,5 +142,197 @@
 ]
 ```
 
+## VibrateCmd
+
+**Description:** Causes a toy that supports vibration to run specific vibration motors at a certain speeds. In order to abstract the dynamic range of different toys, the values are sent as a dictionary of vibration motor indexes against floats with a range of \[0.0-1.0\]
+
+**Introduced In Version:** 1
+
+**Message Version:** 1
+
+**Fields:**
+
+* _Id_ \(unsigned int\): Message Id
+* _DeviceIndex_ \(unsigned int\): Index of device
+* _Speeds_ \(array\): Vibration speeds
+  * _Index_ \(unsigned int\): Index of vibration motor
+  * _Speed_ \(double\): Vibration speed with a range of \[0.0-1.0\]
+
+**Expected Response:**
+
+* Ok message with matching Id on successful request.
+* Error message on value or message error.
+
+**Flow Diagram:**
+
+**Serialization Example:**
+
+```json
+[
+  {
+    "VibrateCmd": {
+      "Id": 1,
+      "DeviceIndex": 0,
+      "Speeds": [
+        {
+          "Index": 0,
+          "Speed": 0.5
+        },
+        {
+          "Index": 1,
+          "Speed": 1
+        }
+      ]
+    }
+  }
+]
+```
+
+## LinearCmd
+
+**Description:** Causes a toy that supports linear movement to reposition it's linear actuators at a certain speeds. In order to abstract the dynamic ranges (both speed and movement) of different toys, the values are sent as a dictionary of linear actuator indexes against objects encapsulating floats with a range of \[0.0-1.0\] for speed and position.
+
+**Introduced In Version:** 1
+
+**Message Version:** 1
+
+**Fields:**
+
+* _Id_ \(unsigned int\): Message Id
+* _DeviceIndex_ \(unsigned int\): Index of device
+* _Vectors_ \(array\): Linear actuator speeds and positions
+  * _Index_ \(unsigned int\): Index of linear actuator
+  * _Speed_ \(double\): Movement speed with a range of \[0.0-1.0\]
+  * _Position_ \(double\): Target position with a range of \[0.0-1.0\]
```

Comment:
Not sure this will work. It leaves the developer blind to the amount of time it would take to get to a position, rendering the message useless for devices that don't give you positional feedback. I would consider something like a modified funscript, where we get position and duration (versus absolute time), and expect the server to calculate speed from there.

Reply:
Doable. This was just really easy to map to the existing command, and I wasn't sure how accurate it would be in practice. I was wondering if distance might be a useful parameter vs position, but it requires us to know the device in a lot of detail.

Reply:
We should probably talk this over with @funjack and maybe mp before implementing.

Reply:
Good plan. Want me to kick that off?